### PR TITLE
[FLINK-5721] Add FoldingState to State Documentation

### PR DESCRIPTION
**Pull request by aljoscha**
R: @NicoK

### CODE REVIEW

File: docs/dev/stream/state.md [hunk 1]
```diff
@@ -113,9 +113,19 @@ be retrieved using `Iterable<T> get()`.
...
+added using `add(T)` are folded into an aggregate using a specified `FoldFunction`.
+
 All types of state also have a method `clear()` that clears the state for the currently
 active key, i.e. the key of the input element.
 
+<span class="label label-danger">Attention</span> `FoldingState` will be deprecated in one of
+the next versions of Flink and will be completely removed in the Future. A more general
+alternative will be provided.
+
+
```
Comment (uce) [inline docs/dev/stream/state.md]: empty line

File: docs/dev/stream/state.md [hunk 2]
```diff
@@ -113,9 +113,19 @@ be retrieved using `Iterable<T> get()`.
...
+* `FoldingState<T, ACC>`: This keeps a single value that represents the aggregation of all values
+added to the state. Contrary to `ReducingState`, the aggregate type is different from the type of
+elements that are added to the state. The interface is the same as for `ListState` but elements
+added using `add(T)` are folded into an aggregate using a specified `FoldFunction`.
+
 All types of state also have a method `clear()` that clears the state for the currently
 active key, i.e. the key of the input element.
 
+<span class="label label-danger">Attention</span> `FoldingState` will be deprecated in one of
+the next versions of Flink and will be completely removed in the Future. A more general
```
Comment (uce) [inline docs/dev/stream/state.md]: Future should be lower case

File: docs/dev/stream/state.md [hunk 3]
```diff
@@ -113,9 +113,19 @@ be retrieved using `Iterable<T> get()`.
 added to the state. The interface is the same as for `ListState` but elements added using
 `add(T)` are reduced to an aggregate using a specified `ReduceFunction`.
 
+* `FoldingState<T, ACC>`: This keeps a single value that represents the aggregation of all values
+added to the state. Contrary to `ReducingState`, the aggregate type is different from the type of
```
Comment (NicoK) [inline docs/dev/stream/state.md]: ...the aggregate type **may be** different from...

File: docs/dev/stream/state.md [hunk 4]
```diff
@@ -126,8 +136,8 @@ To get a state handle, you have to create a `StateDescriptor`. This holds the na
 (as we will see later, you can create several states, and they have to have unique names so
 that you can reference them), the type of the values that the state holds, and possibly
 a user-specified function, such as a `ReduceFunction`. Depending on what type of state you
-want to retrieve, you create either a `ValueStateDescriptor`, a `ListStateDescriptor` or
-a `ReducingStateDescriptor`.
+want to retrieve, you create either a `ValueStateDescriptor`, a `ListStateDescriptor`,
+`ReducingStateDescriptor` or a `FoldingStateDescriptor`.
```
Comment (NicoK) [inline docs/dev/stream/state.md]: **a** `ReducingStateDescriptor`**,** or a ...

**Comment by aljoscha** [comment]
Thanks @uce and @NicoK. I incorporated your suggestions and will merge now.

**Pull request closed without merging**